includes/template-library/manager.php

```diff
@@ -355,6 +355,18 @@ public function get_template_data( array $args ) {
 			return $validate_args;
 		}
 
+		$post_id = intval( $args['template_id'] );
+		$post_status = get_post_status( $post_id );
+		$is_not_admin = ! current_user_can( 'read_private_posts', $post_id );
```

Comment:
```suggestion
		$is_private = 'private' === $post_status && ! current_user_can( 'read_private_posts', $post_id )
```
You only need to perform this check if the post is on private.

Reply:
- `'private' === $post_status && ! current_user_can( 'read_private_posts', $post_id )`
- `'publish' !== $post_status && ! current_user_can( 'edit_post', $post_id )`
- `post_password_required( $post_id ) && ! ( current_user_can( 'edit_post', $post_id ) || current_user_can( 'administrator' ) ) && ! isset( $_COOKIE[ 'wp-postpass_' . COOKIEHASH ] ) )`

I believe these are the checks you should have. The third one includes @mykytamurzin's concern about admin should be allowed even if password protected.

`if ( $is_private || $not_allowed || $is_password_protected ) {...}`